lib/categories.nix

```diff
@@ -0,0 +1,949 @@
+/* List of categories
+    ```nix
+    handle = {
+      # Required
+      name = "Category Name";
+      description = ''
+        Description of category
+      '';
+      relatedCategories = [ related01 related02 . . . ];
+    };
+    ```
+
+    where
+
+    - `handle` is the handle you are going to use in nixpkgs expressions;
+       usually the same as the name but in camelCase
+    - `name` is the category name
+    - `description` is a description of the category
+    - `relatedCategories` is a (possibly empty) list of related categories (like
+      the `handle` above)
+
+    More fields may be added in the future.
+
+    When editing this file:
+     * do not modify any of
+         - mainCategories
+         - additionalCategories
+         - reservedCategories
+       except by reasons of force majeure (e.g. an update on the Freedesktop.org
+       Desktop Menu Specification);
+     * keep the lists alphabetically sorted;
+     * test the validity of the format with:
+         nix-build lib/tests/categories.nix
+*/
+{ lib }:
+
+let
+  mainCategories = {
```

Comment:
I think we can remove the `mainCategories`/`additionalCategories` distinction, especially if we establish a hierarchy as described in https://github.com/NixOS/nixpkgs/pull/325357/files#r1668350336. Furthermore, RFC146 (section "Semantic Details") establishes "most important" categories to be a package-level concept.

Reply:
The intention is not to establish a hierarchy, but to reflect Appstream.

Reply:
A priori, for automatic conversion one doesn't need all FreeDesktop.org categories to have different images…

Reply:
"images"?

Reply:
In the sense of a function mapping preimages to images. Automatically deciding something based on FreeDesktop.org categories might be useful even we decide the same thing for specific different categories.

Reply:
Image in the sense of mathematical functions? "We do not need a bijective mapping between Appstream and Nix Categories"?

Reply:
Yes.

We _cannot_ have a surjective mapping (which would mean no non-FreeDesktop.org-matching categories) because Appstream is about apps and we have some very non-apps packages (and Freedesktop.org categorisation is for application launchers even before Appstream).

We are not obliged to have an injective mapping either, as some distinctions can be ignored or not ignored as a policy decision.